src/nft/components/card/media.tsx

```diff
@@ -0,0 +1,329 @@
+
+  ${StyledImageContainer}:hover & {
+    display: block;
+  }
+`
+
+const StyledVideo = styled.video<{
+  $aspectRatio?: string
+}>`
+  width: 100%;
+  aspect-ratio: ${({ $aspectRatio }) => $aspectRatio};
+`
+
+const StyledInnerMediaContainer = styled(Row)`
+  position: absolute;
+  left: 0px;
+  top: 0px;
+`
+
+export const NftVideo = ({
+  src,
+  mediaSrc,
+  tokenId,
+  uniformAspectRatio = UniformAspectRatios.square,
+  setUniformAspectRatio,
+  renderedHeight,
+  setRenderedHeight,
+  shouldPlay,
+  setCurrentTokenPlayingMedia,
+}: MediaProps & ImageProps) => {
+  const vidRef = useRef<HTMLVideoElement>(null)
+  const [noContent, setNoContent] = useState(!src)
+  const [imageLoaded, setImageLoaded] = useState(false)
+
+  useEffect(() => {
+    if (shouldPlay && vidRef.current) {
+      vidRef.current.play()
+    } else if (!shouldPlay && vidRef.current) {
+      vidRef.current.pause()
+    }
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [shouldPlay, vidRef.current])
+
+  if (noContent) {
+    return <NoContentContainer height={getHeightFromAspectRatio(uniformAspectRatio, renderedHeight)} />
+  }
+
+  return (
+    <>
+      <StyledMediaContainer>
+        <StyledImage
+          src={src}
+          $aspectRatio={getMediaAspectRatio(uniformAspectRatio, setUniformAspectRatio)}
+          imageLoading={!imageLoaded}
+          draggable={false}
+          onError={() => setNoContent(true)}
+          onLoad={(e) => {
+            handleUniformAspectRatio(uniformAspectRatio, e, setUniformAspectRatio, renderedHeight, setRenderedHeight)
+            setImageLoaded(true)
+          }}
+          $hidden={shouldPlay}
+        />
+      </StyledMediaContainer>
+      {shouldPlay ? (
+        <>
+          <PlaybackButton pauseButton={true}>
+            <Pause
+              size="24px"
+              onClick={(e) => {
+                e.preventDefault()
+                e.stopPropagation()
+                setCurrentTokenPlayingMedia(undefined)
+              }}
+            />
+          </PlaybackButton>
+          <StyledInnerMediaContainer>
+            <StyledVideo
+              $aspectRatio={getMediaAspectRatio(uniformAspectRatio, setUniformAspectRatio)}
+              ref={vidRef}
+              onEnded={(e) => {
+                e.preventDefault()
+                setCurrentTokenPlayingMedia(undefined)
+              }}
+              loop
+              playsInline
+            >
+              <source src={mediaSrc} />
+            </StyledVideo>
+          </StyledInnerMediaContainer>
+        </>
+      ) : (
+        <PlaybackButton>
+          <Play
+            size="24px"
+            onClick={(e) => {
+              e.preventDefault()
+              e.stopPropagation()
+              setCurrentTokenPlayingMedia(tokenId)
+            }}
+          />
+        </PlaybackButton>
+      )}
+    </>
+  )
+}
+
+const StyledAudio = styled.audio`
+  width: 100%;
+  height: 100%;
+`
+
+export const NftAudio = ({
```

Comment:
NftAudio and NftVideo are nearly identical except for whats inside `StyledVideo/StyledAudio` can we condense?